tests_e2e/tests/agent_status/agent_status.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+
+# Microsoft Azure Linux Agent
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# Validates the agent status is updated without any other goal state changes
+#
+
+import datetime
+import json
+
+from azure.mgmt.compute.models import VirtualMachineInstanceView
+from assertpy import assert_that
+from time import sleep
+
+from tests_e2e.tests.lib.agent_test import AgentTest
+from tests_e2e.tests.lib.agent_test_context import AgentTestContext
+from tests_e2e.tests.lib.logging import log
+from tests_e2e.tests.lib.virtual_machine_client import VirtualMachineClient
+
+
+class AgentStatus(AgentTest):
+    def __init__(self, context: AgentTestContext):
+        super().__init__(context)
+        self._ssh_client = self._context.create_ssh_client()
+
+    def validate_instance_view_vmagent_status(self, instance_view: VirtualMachineInstanceView):
+        status = instance_view.vm_agent.statuses[0]
+
+        # Validate message field
+        message = status.message
+        if message is None:
+            raise Exception("Instance view is missing an agent status message, waiting to retry...")
```

Comment:
waiting to retry, bit misleading. Looks like this exception is logging after retries completed when we log timeout expired.